inc/packages/namespace.php

```diff
@@ -0,0 +1,580 @@
+		$error = sprintf( 'The given package filepath %s does not exist.', $file );
+		return new WP_Error( 'altis_search_file_not_found', $error );
+	}
+
+	// Default package ID is the full path to the package.
+	// Note this will only work if Elasticsearch is running on the same server as WordPress.
+	$package_id = $file;
+
+	/**
+	 * Override the package ID on creation for a given package file name.
+	 *
+	 * @param string|WP_Error The package ID for the given file.
+	 * @param string $slug The package slug.
+	 * @param string $file The full package file path.
+	 * @param bool $for_network True if the package is a network level package.
+	 */
+	$package_id = apply_filters( 'altis.search.create_package_id', $package_id, $slug, $file, $for_network );
+	if ( is_wp_error( $package_id ) ) {
+		if ( $for_network ) {
+			update_site_option( "altis_search_package_status_{$slug}", 'ERROR' );
+		} else {
+			update_option( "altis_search_package_status_{$slug}", 'ERROR' );
+		}
+		return $package_id;
+	}
+
+	// Store package ID.
+	if ( $for_network ) {
+		update_site_option( "altis_search_package_{$slug}", $package_id );
+	} else {
+		update_option( "altis_search_package_{$slug}", $package_id );
+	}
+
+	/**
+	 * Action triggered when a new package has been created.
+	 *
+	 * Default usage is to trigger an update to the ES mapping.
+	 *
+	 * @param string $package_id The package ID.
+	 * @param string $slug The package slug.
+	 * @param bool $for_network True if the package was created at the network level.
+	 */
+	do_action( 'altis.search.created_package', $package_id, $slug, $for_network );
+
+	// Return the reference used to get the package in an analyzer.
+	return $package_id;
+}
+
+/**
+ * Removes the stored package ID and file for a given package.
+ *
+ * @param string $slug The package file path.
+ * @param bool $for_network If true deletes the network level package.
+ * @return bool|WP_Error
+ */
+function delete_package( string $slug, bool $for_network = false ) {
+	// Get the package ID to pass to action hook.
+	$package_id = get_package_id( $slug, $for_network );
+	if ( empty( $package_id ) ) {
+		return new WP_Error(
+			'delete_package_error',
+			// translators: %s replaced by package slug.
+			sprintf( __( 'Package for slug %s does not exist', 'altis' ), $slug )
+		);
+	}
+
+	$file = get_package_path( $slug, $for_network );
+
+	// Delete the file.
+	unlink( $file );
+
+	// Remove the stored package ID.
+	if ( $for_network ) {
+		$deleted = delete_site_option( "altis_search_package_{$slug}" );
+		delete_site_option( "altis_search_package_status_{$slug}" );
+		delete_site_option( "altis_search_package_error_{$slug}" );
+	} else {
+		$deleted = delete_option( "altis_search_package_{$slug}" );
+		delete_option( "altis_search_package_status_{$slug}" );
+		delete_option( "altis_search_package_error_{$slug}" );
+	}
+
+	// Update the index settings on successful deletion.
+	if ( $deleted ) {
```

Comment:
Should be flipped so that the happy path is the main one (and there's less code indented then too, which is #aesthetics)